### ISSUE TITLE
Alternative note names

### ISSUE THREAD

**Issue by jmmelko** [sky-music/sky-python-music-sheet-maker#70]
- [x] doremi: ti is sometimes used instead of si
- [ ] western CDEFGAB: German use H instead of B, and B instead of Bb

Can we do something?

**Comment by jmmelko** [comment]
I have added ti in the beta. This should do no harm and we are nobody to judge wether ti is bad usage or not.

However, for B and H, handling them would be clumsy. Also I have read in Wikipedia that many people from Germanic countries will use the English system when writing notes, but reading them out loud using their own notation. So, as long as the bot does not speak, we can keep it like this.

**Comment by t1-tracey** [comment]
This is interesting, thanks for bringing it up - I didn’t know that about German. We could handle it by adding a dictionary in the Western note parser:

```

self.GERMAN_CHROMATIC_SCALE_DICT = {'C': 0, 'C#': 1, 'Db': 1, 'D': 2, 'D#': 3, 'Eb': 3, 'E': 4, 'F': 5, 'F#': 6, 'Gb': 6, 'G': 7, 'G#': 8, 'Ab': 8, 'A': 9, 'A#': 10, 'B': 10, 'H': 11}

```

And we would switch to that when `get_chromatic_scale_dict()` is called. (Need a way to know if the user wants to use it though hmm, don’t have an idea on how to approach that yet, after what you mentioned about `os`)

And yeah I don’t know how frequently it might be used, I do have a friend I can ask though

**Comment by t1-tracey** [comment]
And lol I do use ti, (do re mi fa so la ti do) I hadn’t realised it was different in French 😃 thanks for adding it.

**Comment by jmmelko** [comment]
> This is interesting, thanks for bringing it up - I didn’t know that about German. We could handle it by adding a dictionary in the Western note parser:
> 
> ```
> 
> self.GERMAN_CHROMATIC_SCALE_DICT = {'C': 0, 'C#': 1, 'Db': 1, 'D': 2, 'D#': 3, 'Eb': 3, 'E': 4, 'F': 5, 'F#': 6, 'Gb': 6, 'G': 7, 'G#': 8, 'Ab': 8, 'A': 9, 'A#': 10, 'B': 10, 'H': 11}
> ```
> 
> And we would switch to that when `get_chromatic_scale_dict()` is called. (Need a way to know if the user wants to use it though hmm, don’t have an idea on how to approach that yet, after what you mentioned about `os`)
> 
> And yeah I don’t know how frequently it might be used, I do have a friend I can ask though

I think we can keep it mind that’s possible but let it rest for a moment, until someone complains.
I have read that Germans are not the only ones to use it but also danish, Finnish, norvegians, etc.

**Comment by t1-tracey** [comment]
I see, ok :) I’m good with leaving it till someone complains

Sent from my iPad

On 21 Dec 2019, at 3:52 am, jmmelko <notifications@github.com> wrote:

﻿

Closed #70<https://github.com/sky-music/sky-python-music-sheet-maker/issues/70>.

—
